### PR TITLE
Memory buildup clean

### LINKED ISSUE #157: Small memory buildup
When doing very long runs, the memory footprint increases, which ends up crashing the run. Specially since we [only save the weights](https://github.com/Clay-foundation/model/blob/3f4210ff3160240e73c4d3541962032afef957db/trainer.py#L50) this could be a problem for future long runs.

The plot below comes from the same run as #156 

Screenshot 2024-02-18 at 23 33 20

### PR DESCRIPTION
**Pull request by brunosan**
fixes #157.

Using malloc I tested that the biggest mem footprint buildup after n epochs is from matplotlib, which we use on wandb plots. 

Just closing the plot frees most of the buildup.

```logs
1 epochs 
matplotlib/cbook.py:733: size=32.0 MiB, count=66, average=497 KiB
<frozen importlib._bootstrap_external>:729: size=2997 KiB, count=27469, average=112 B
matplotlib/transforms.py:198: size=1515 KiB, count=16624, average=93 B
matplotlib/lines.py:359: size=1336 KiB, count=1728, average=792 B

100 epochs before fix
matplotlib/cbook.py:733: size=1616 MiB, count=3333, average=497 KiB
matplotlib/transforms.py:198: size=74.7 MiB, count=839610, average=93 B
matplotlib/lines.py:359: size=65.9 MiB, count=87264, average=792 B
matplotlib/text.py:994: size=61.0 MiB, count=80800, average=792 B

100 epochs clearing "all" plots
matplotlib/cbook.py:733: size=64.0 MiB, count=132, average=497 KiB
matplotlib/transforms.py:198: size=3031 KiB, count=33252, average=93 B
<frozen importlib._bootstrap_external>:729: size=2995 KiB, count=27459, average=112 B
matplotlib/lines.py:359: size=2673 KiB, count=3456, average=792 B

100 epochs clearing fig plot
matplotlib/cbook.py:733: size=32.0 MiB, count=66, average=497 KiB
<frozen importlib._bootstrap_external>:729: size=2988 KiB, count=27324, average=112 B
matplotlib/transforms.py:198: size=1515 KiB, count=16626, average=93 B
matplotlib/lines.py:359: size=1336 KiB, count=1728, average=792 B

```

I left the malloc code commented out to make it easer in the future if needed.

### CODE REVIEW

File: src/callbacks_wandb.py [hunk 1]
```diff
@@ -235,3 +235,4 @@ def on_validation_end(
                 axs[1, i].axis("off")
 
             self.logger.experiment.log({"Images": wandb.Image(fig)})
+            plt.close("all")
```
Comment (yellowcap) [inline src/callbacks_wandb.py:238]: Could we do `plt.close(fig)` here to be more explicit what we are closing?

File: trainer.py [hunk 2]
```diff
@@ -74,5 +75,12 @@ def cli_main(
 
 # %%
 if __name__ == "__main__":
+    # tracemalloc.start()
     cli_main()
+
+    # snapshot = tracemalloc.take_snapshot()
```
Comment (yellowcap) [inline trainer.py:81]: This commented out tracing code should be removed before merging.

**Review by yellowcap — CHANGES_REQUESTED** [review]
Good catch @brunosan . Made two small suggestions to edit before merging.

File: src/callbacks_wandb.py (hunk 1 above)
Comment (brunosan) [reply src/callbacks_wandb.py]: I was trying to be extra cautious and class everything, but I´ve tested and there´s no difference. Changed.

File: trainer.py (hunk 2 above)
Comment (brunosan) [reply trainer.py]: Done

**Review by yellowcap — APPROVED** [review]
Looks good

**Pull request merged**